### ISSUE TITLE
Accessibility concerns on mdn-bcd-collector website

### ISSUE THREAD

**Issue by tannerdolby** [foolip/mdn-bcd-collector#1410]
After running a [Lighthouse](https://developers.google.com/web/tools/lighthouse) audit on https://mdn-bcd-collector.appspot.com/, we receive this audit report:

CC @vinyldarkscratch 

<img width="735" alt="Screen Shot 2021-08-17 at 3 46 39 PM" src="https://user-images.githubusercontent.com/48612525/129810534-0f65ef0d-7316-4360-8068-c2120b27e941.png">

The Performance and and Best Practices scores are great (performance could be improved), but the Accessibility and SEO definitely need some work. Contrast ratios failing WCAG 2.0 and images with alt text, along with a few other things. They are all quick fixes so I will pepper a single PR with the majority of the changes.

## Accessibility report
Issues needing to be fixed:
- Background and foreground colors do not have a sufficient contrast ratio.
- Image elements do not have [alt] attributes
- `<html>` element does not have a [lang] attribute
<img width="734" alt="Screen Shot 2021-08-17 at 4 01 31 PM" src="https://user-images.githubusercontent.com/48612525/129811575-a12cec5d-bbd9-4a80-a4d1-f47ccfd22434.png">

## SEO report
Issues needing to be fixed:
- Document does not have a meta description
- Image elements do not have [alt] attributes

<img width="736" alt="Screen Shot 2021-08-17 at 4 01 51 PM" src="https://user-images.githubusercontent.com/48612525/129811603-4f7fcd49-83a5-4244-a447-098129135282.png">

**Comment by tannerdolby** [comment]
According to the Lighthouse documentation - https://web.dev/color-contrast/?utm_source=lighthouse&utm_medium=devtools

- Text that is 18 pt, or 14 pt and bold, needs a contrast ratio of 3:1.
- All other text needs a contrast ratio of 4.5:1.

> WCAG 2.0 level AA requires a contrast ratio of at least 4.5:1 for normal text and 3:1 for large text. https://webaim.org/resources/contrastchecker/

The foreground text being `#fff` on a `#3D8A3F` background creates a 4:28:1 contrast ratio, which since the text is 14px and not bold it doesn't pass WCAG 2.1 since its not 18px size or 14px bold. Therefore, even though it exceeds 3.5:1 it doesn't meet either of the reqs (18px and 14px bold) so it doesn't meet the 4.5:1 expectation, thus Lighthouse throwing an issue.

If adjust the color of the background to be `#39843B` then the contrast ratio exceeds 4.5:1 and Lighthouse won't complain about contrast ratios. Or make the 14px text bold so it qualifies for the 3.5:1 ratio req.

Check contrast ratios here: https://webaim.org/resources/contrastchecker/

**Comment by foolip** [comment]
Thanks for the repo, @tannerdolby!

@vinyldarkscratch can you look into the color contrast issues? The fixes in https://github.com/foolip/mdn-bcd-collector/pull/1411 look good to me.

**Comment by foolip** [comment]
Reopening since the color contrast issues remain.

**Comment by vinyldarkscratch** [comment]
Thanks for the PR containing fixes!  I feel embarrassed for not adding `alt` tags to the images...  😅 

Regarding the website's accessibility, front-end UI and UX development hasn't been a strong focus since the initial design work, and much more on the backend and test code.  Since this project is for a very select audience (specifically BCD's contributors), adjustments designed to aid in usage by the general public hasn't been a priority of ours.  Nevertheless, we're happy to accept PRs that make such adjustments!

With that said, it'll probably be a while before I get to adjusting the color contrast for further accessibility.  Our goals for this quarter involve getting BCD's real value count as high as we can, so my attention will be directed towards updating the custom tests and synchronizing results.  If you (or someone) write(s) a PR improving the color contrast before I do, I'm more than happy to review it!

**Comment by tannerdolby** [comment]
> Thanks for the repo, @tannerdolby!

Your welcome @foolip! I'm happy to update the contrast ratios as well.

**Comment by tannerdolby** [comment]
> Thanks for the PR containing fixes! I feel embarrassed for not adding alt tags to the images... 😅

No problem @vinyldarkscratch! Don't be embarrassed it happens to the best of us 😄. Those images are "close to decorative" so doing `<img alt="">` fixes the Lighthouse warning by removing them from the accessibility tree. But I think giving them appropriate `alt` text like `Icon for Safari browser` etc would be better as they do have some meaning representing browser compatibility, now that I have a second look at the HTML.

**Comment by tannerdolby** [comment]
>  If you (or someone) write(s) a PR improving the color contrast before I do, I'm more than happy to review it!

Ok sounds good. I will open a separate PR later today to add appropriate alt text to those browser compatibility images, along with fixing the contrast ratios.

**Comment by vinyldarkscratch** [comment]
Leaving the alt text blank would be better in my opinion -- if we fill it with something, screen readers will read off their text, and since the images are helper icons that go along with existing text, this works perfectly.  (Otherwise, the screen readers would read the page as "Chrome Chrome 1 plus" or something like that.)

Looking forward to your next PR, thank you!

**Comment by tannerdolby** [comment]
> Leaving the alt text blank would be better in my opinion -- if we fill it with something, screen readers will read off their text, and since the images are helper icons that go along with existing text, this works perfectly.

Agreed! 👍 

And excellent, I should submit it sometime this afternoon!

**Comment by tannerdolby** [comment]
While I'm at it, the `:hover` state for `button, select, input` creates a contrast ratio of 2.77:1 using `#4CAF50` as background with `#fff` foreground text, which I'll update as well. Although these are interactive elements, the same WCAG 2.1 contrast ratio rules apply to them.

> Color changes on hover, focus, etc: Text sometimes changes color while the user interacts with it using a mouse or keyboard. CSS can be used to define hover, focus, or active states for interactive elements. There is no mention of special consideration for these changes in text color, so text in all these states must meet the same contrast requirements (and must be evaluated independently). - https://webaim.org/articles/contrast/

I was able to decrease the lightness by using `#48A34B` with `#fff` foreground text and that yields 3.18:1 which atleast passes WCAG AA for large text (which its not). Moving up to 4.5:1 was a challenge to find that same sort of "brighter" hover state background, but it seems to look quite nice with `#0F8A13` which has 4.5:1 meeting WCAG AA for normal text. This is why some designers/devs aim for a "darker" background-color on hover for interactive elements, but either way is still a contrast battle.

Also, the `input::placeholder` text color of `#ccc` on the green background creates another really low contrast ratio of 2.82:1. I understand we want this to look "lighter" than its neighboring interactive elements but it still needs to be legible for low-vision users.
